Add alarm clusters support

elemoine opened the pull request:
This introduces support for alarm clusters. WIP at the moment.

Commit:
Define the highest_severity policy

Commit:
Generate the gse_policies Lua module

Commit:
Generate gse topology module for each alarm cluster

Commit:
Generate gse filter toml for each cluster alarm

Commit:
Adapt GSE Lua code

SwannCroiset commented on heka/files/lua/common/gse.lua:
```diff
@@ -36,10 +36,10 @@ local ordered_clusters = {}
 
```
I would really like to have Lua unittests. What about the feature branch **mkxx** on legacy repository ?

elemoine replied:
I understand your concerns. Thinking about it I am not sure creating an mkxx branch in fuel-plugin-lma-collector makes sense, but let's discuss it.

elemoine replied:
https://github.com/tcpcloud/salt-formula-heka/pull/32 adds a script for running Lua tests, and all the Lua tests from fuel-plugin-lma-collector.

elemoine replied:
#32 is merged, and this PR changes some of the tests.

Commit:
Remove gse cluster_field parameter

This parameter is not needed anymore. Heka's message_matchers are now used to
match input messages.

Commit:
Support dimensions in gse metrics

Commit:
Do not rely on pacemaker_local_resource_active

elemoine commented on heka/_service.sls:
```diff
@@ -279,6 +283,60 @@ heka_{{ service_name }}_grain:
+  - mode: 640
+  - group: heka
+  - require:
+    - file: heka_{{ service_name }}_conf_dir
```
Change this, the file is not in the conf dir.

elemoine commented on heka/_service.sls:
```diff
@@ -279,6 +283,60 @@ heka_{{ service_name }}_grain:
+  - mode: 640
+  - group: heka
+  - require:
+    - file: heka_{{ service_name }}_conf_dir
```
Change this, the file is not in the conf dir.

elemoine commented on heka/files/toml/filter/gse_alarm_cluster.toml:
```diff
@@ -0,0 +1,11 @@
+
+[gse_{{ alarm_cluster_name }}_filter.config]
+topology_file = "gse_{{ alarm_cluster_name|replace('-', '_') }}_topology"
+dimensions = '{{ salt['heka_alarming.dimensions'](alarm_cluster)|json }}'
```
Add `interval`, `warm_up_period`, etc. (all the config options supported by the `gse_cluster_filter`).

elemoine commented on heka/meta/heka.yml:
```diff
@@ -108,6 +108,41 @@ remote_collector:
+            arguments: [ okay ]
+            relational_operator: '>'
+            threshold: 0
+    - status: unknown
```
Add the other built-in policies.

Commit:
Define the majority_of_members policy

Commit:
Define the availability_of_members policy

elemoine commented:
This is now ready for review. What I have not tested up to now is clusters of clusters.
